doc/devel/release_guide.rst

```diff
@@ -1,169 +1,250 @@
-    git tag -a v1.1.0rc1
+Congratulations, the scariest part is done!
 
-The `-a` flag will allow you to write a message about the tag, and
-affiliate your name with it. A reasonable tag message would be something
-like ``v1.1.0 Release Candidate 1 (September 24, 2011)``. To tag a
-release after the fact, just track down the commit hash, and::
+To prevent issues with any down-stream builders which download the
+tarball from github it is important to move all branches away from the commit
+with the tag ::
 
-    git tag -a v1.0.1rc1 a9f3f3a50745
+  git commit --allow-empty
+  git push DANGER master
 
-Tags allow developers to quickly checkout different releases by name,
-and also provides source download via zip and tarball on github.
+If this is a final release, also create a 'doc' branch (this is not
+done for pre-releases)::
 
-Then push the tags to the main repository::
+   git branch v2.0.0-doc
+   git push DANGER v2.0.0-doc
 
-    git push upstream v1.0.1rc1
+and if this is a major or minor release, also create a bug-fix branch (a
+micro release will be cut off of this branch)::
 
-.. _release-packaging:
+   git branch v2.0.x
+   git push DANGER v2.0.x
 
-Packaging
-=========
 
-* Make sure the :file:`MANIFEST.in` is up to date and remove
-  :file:`MANIFEST` so it will be rebuilt by MANIFEST.in
 
-* run `git clean` in the mpl git directory before building the sdist
+.. _release_tag:
 
-* unpack the sdist and make sure you can build from that directory
+Release Management / DOI
+------------------------
 
-* Use :file:`setup.cfg` to set the default backends.  For windows and
-  OSX, the default backend should be TkAgg.  You should also turn on
-  or off any platform specific build options you need.  Importantly,
-  you also need to make sure that you delete the :file:`build` dir
-  after any changes to :file:`setup.cfg` before rebuilding since cruft
-  in the :file:`build` dir can get carried along.
+Via the github UI (chase down link), turn the newly pushed tag into a
+release.  If this is a pre-release remember to mark it as such.
 
-* On windows, unix2dos the rc file.
+For final releases also get a DOI from `zenodo
+<https://zenodo.org/>`__ and edit :file:`doc/_templates/citing.html`
+with DOI link and commit to the VER-doc branch and push to github ::
 
-* We have a Makefile for the OS X builds in the mpl source dir
-  :file:`release/osx`, so use this to prepare the OS X releases.
+  git push DANGER v2.0.0-doc:v2.0.0-doc
 
-* We have a Makefile for the win32 mingw builds in the mpl source dir
-  :file:`release/win32` which you can use this to prepare the windows
-  releases.
+.. _release_bld_bin:
 
+Building binaries
+-----------------
 
-Update PyPI
-===========
+We distribute mac, windows, and many linux wheels as well as a source
+tarball via pypi.  Before uploading anything, contact the various
+builders.  Mac and manylinux wheels are built on travis
+.  You need to edit the
+:file:`.travis.yml` file and push to master of `the build
```

Comment:
I'd say "... `.travis.yml` file on the master branch of ...". Otherwise, it sounds like you have to push a copy of `matplotlib` itself to that repo.